### PR TITLE
Have TestHost populate features before calling IHttpContextFactory.Create

### PR DESCRIPTION
**Pull request by Tratcher**
#896 IHttpContextFactory.Create may want to act on the values.

### CODE REVIEW

File: src/Microsoft.AspNetCore.TestHost/ClientHandler.cs [hunk 1]
```diff
@@ -126,49 +126,51 @@ internal RequestState(HttpRequestMessage request, PathString pathBase, IHttpAppl
...
-                Context = application.CreateContext(contextFeatures);
-                var httpContext = Context.HttpContext;
+                var requestLifetimeFeature = new HttpRequestLifetimeFeature();
+                contextFeatures.Set<IHttpRequestLifetimeFeature>(requestLifetimeFeature);
 
-                var serverRequest = httpContext.Request;
-                serverRequest.Protocol = "HTTP/" + request.Version.ToString(2);
-                serverRequest.Scheme = request.RequestUri.Scheme;
-                serverRequest.Method = request.Method.ToString();
+                requestFeature.Protocol = "HTTP/" + request.Version.ToString(2);
```
Comment (cesarbs) [inline src/Microsoft.AspNetCore.TestHost/ClientHandler.cs]: I know it was already there, but what does the 2 stand for here?

Comment (Tratcher) [inline src/Microsoft.AspNetCore.TestHost/ClientHandler.cs]: two segments (1.0) rather than the full 4 segments.

Comment (cesarbs) [inline src/Microsoft.AspNetCore.TestHost/ClientHandler.cs]: Ok. I think switching to a named parameter makes it clearer.

File: test/Microsoft.AspNetCore.TestHost.Tests/ClientHandlerTests.cs [hunk 2]
```diff
@@ -50,6 +50,30 @@ public Task ExpectedKeysAreAvailable()
...
+                Assert.Equal("/A/Path", features.Get<IHttpRequestFeature>().PathBase);
+                Assert.Equal("/and/file.txt", features.Get<IHttpRequestFeature>().Path);
+                Assert.Equal("?and=query", features.Get<IHttpRequestFeature>().QueryString);
+                Assert.NotNull(features.Get<IHttpRequestFeature>().Body);
+                Assert.NotNull(features.Get<IHttpRequestFeature>().Headers);
+                Assert.NotNull(features.Get<IHttpResponseFeature>().Headers);
+                Assert.NotNull(features.Get<IHttpResponseFeature>().Body);
+                Assert.Equal(200, features.Get<IHttpResponseFeature>().StatusCode);
+                Assert.Null(features.Get<IHttpResponseFeature>().ReasonPhrase);
+                Assert.Equal("example.com", features.Get<IHttpRequestFeature>().Headers["host"]);
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.TestHost.Tests/ClientHandlerTests.cs]: Also check `IHttpRequestLifetimeFeature` is not null?

**Pull request merged**